### PR TITLE
Refactor for advanced use cases

### PR DESCRIPTION
**Pull request by GrahamTheCoder**
This is some minor tidying up that I wanted to do before tackling some more [advanced use cases](https://github.com/BenPhegan/NuGet.Extensions/pull/37).
I don't believe it changes any behaviour someone would reasonably be depending on.

### CODE REVIEW

**Comment by GrahamTheCoder** [comment]
I've rebased this onto master

File: NuGet.Extensions/ReferenceAnalysers/ProjectNugetifier.cs [hunk 1]
```diff
@@ -31,7 +31,7 @@ public ProjectNugetifier(IVsProject vsProject, IPackageRepository packageReposit
 
         public ICollection<IPackage> NugetifyReferences(DirectoryInfo solutionDir)
         {
-            var binaryReferences = _vsProject.GetBinaryReferences().ToList();
+            var binaryReferences = GetReferences().ToList();
```
Comment (BenPhegan) [inline NuGet.Extensions/ReferenceAnalysers/ProjectNugetifier.cs]: I assume this is used in subsequent PRs?

**Comment by BenPhegan** [comment]
Looks fine, sorry for delay!

File: NuGet.Extensions/ReferenceAnalysers/ProjectNugetifier.cs (hunk 1 above)
Comment (GrahamTheCoder) [inline NuGet.Extensions/ReferenceAnalysers/ProjectNugetifier.cs]: Yep it's used in the first commit of PR https://github.com/BenPhegan/NuGet.Extensions/pull/37

**Pull request merged**